R/nbhd_ses_variables.R

```diff
@@ -0,0 +1,121 @@
+
+                                          from temp_data tmp
+                                          
+                                          left join locality_variables l
+                                            on l.genc_id = tmp.genc_id
+                                          left join lookup_statcan_v2016 s
+                                            on l.da16uid = s.da16uid; " 
+                               
+    ) %>% as.data.table()
+    # cal % visible minority
+    var_tbl[, c16_vismin_pct := round(c16_vismin / (c16_vismin + c16_vismin_not) * 100, 2 )]
+    
+    # cal % immigration status
+    var_tbl[, c16_immsta_pct := round(c16_immsta_imm / c16_immsta * 100, 2 )]
+    
+    # cal % post-secondary education (>15)
+    var_tbl[, c16_ed_15over_postsec_pct := round(c16_ed_15over_postsec / c16_ed_15over * 100, 2 )]
+    
+    # cal % post-secondary education (25-64)
+    var_tbl[, c16_ed_25to64_postsec_pct := round(c16_ed_25to64_postsec / c16_ed_25to64 * 100, 2 )]
+    
+    var_tbl <- var_tbl[, !c('c16_vismin', 'c16_vismin_not', 'c16_immsta_imm', 'c16_immsta', 'c16_ed_15over_postsec', 'c16_ed_15over', 'c16_ed_25to64_postsec', 'c16_ed_25to64'
+    )]  
+    
+    
+  }
+  
+  if  (dissemination_area==2021) {
+    var_tbl <- DBI::dbGetQuery(dbcon, 'select tmp.genc_id, l.da21uid, s.c21_vismin, s.c21_vismin_not, s.qnatippe, s.qnbtippe, s.qaatippe, s.qabtippe, s.atippe, s.btippe, s.c21_immsta, s.c21_immsta_imm, s.c21_ed_15over_postsec, s.c21_ed_15over, s.c21_ed_25to64_postsec, s.c21_ed_25to64, "households_dwellings_DA21", "material_resources_DA21", "age_labourforce_DA21", "racialized_NC_pop_DA21", "households_dwellings_q_DA21", "material_resources_q_DA21", "age_labourforce_q_DA21", "racialized_NC_pop_q_DA21"
+
+                                          from temp_data tmp
+                                          
+                                          left join locality_variables l
+                                            on l.genc_id = tmp.genc_id
+                                          left join lookup_statcan_v2021 s
+                                            on l.da21uid = s.da21uid; ' 
+                               
+    ) %>% as.data.table()
+    
+    # cal % visible minority
+    var_tbl[, c21_vismin_pct := round(c21_vismin / (c21_vismin + c21_vismin_not) * 100, 2 )]
+    
+    # cal % immigration status
+    var_tbl[, c21_immsta_pct := round(c21_immsta_imm / c21_immsta * 100, 2 )]
+    
+    # cal % post-secondary education (>15)
+    var_tbl[, c21_ed_15over_postsec_pct := round(c21_ed_15over_postsec / c21_ed_15over * 100, 2 )]
+    
+    # cal % post-secondary education (25-64)
+    var_tbl[, c21_ed_25to64_postsec_pct := round(c21_ed_25to64_postsec / c21_ed_25to64 * 100, 2 )]
+    
+    var_tbl <- var_tbl[, !c('c21_vismin', 'c21_vismin_not', 'c21_immsta_imm', 'c21_immsta', 'c21_ed_15over_postsec', 'c21_ed_15over', 'c21_ed_25to64_postsec', 'c21_ed_25to64'
+    )] 
```

Comment:
This section is a little bit redundant because the calculations are the same as for the 2016 census data. The only thing that's different is the prefix (`c21` vs. `c16`) in the variable names, so you could potentially consider calculating this outside of the if statement.